src/protocol.cpp

```diff
@@ -2090,6 +2113,40 @@ bool CProtocol::EvaluateCLChannelLevelListMes  ( const CHostAddress&     InetAdd
     return false; // no error
 }
 
+void CProtocol::CreateCLRegisterServerResp  ( const CHostAddress& InetAddr,
+                                              const int           iStatus )
+{
+    int              iPos = 0; // init position pointer
+    CVector<uint8_t> vecData( 1 );
+
+    PutValOnStream ( vecData, iPos,
+        static_cast<uint32_t> ( iStatus ), 1 );
+
+    CreateAndImmSendConLessMessage ( PROTMESSID_CLM_REGISTER_SERVER_RESP,
+                                     vecData,
+                                     InetAddr );
+}
+
+bool CProtocol::EvaluateCLRegisterServerResp ( const CHostAddress&     InetAddr,
+                                               const CVector<uint8_t>& vecData )
+{
+    int       iPos     = 0; // init position pointer
+    const int iDataLen = vecData.Size();
+
+    if ( iDataLen != 1 )
+    {
+        return true;
+    }
+
+    // known values are 0 and 1 currently
+    // but allow 0 to 255
+    int iStatus = static_cast<int> ( GetValFromStream ( vecData, iPos, 1 ) );
```

Comment:
Retrieve as int, validate that the value is positive and in range for the enum, then cast.